interfaces/builtin/media_hub.go

```diff
@@ -0,0 +1,190 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2017 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package builtin
+
+import (
+	"strings"
+
+	"github.com/snapcore/snapd/interfaces"
+	"github.com/snapcore/snapd/interfaces/apparmor"
+)
+
+const mediaHubPermanentSlotAppArmor = `
+# Description: Allow operating as the the media-hub service.
+
+# DBus accesses
+#include <abstractions/dbus-session-strict>
+#include <abstractions/dbus-strict>
+
+dbus (send)
+    bus=session
+    path=/org/freedesktop/DBus
+    interface=org.freedesktop.DBus
+    member="{Request,Release}Name"
+    peer=(name=org.freedesktop.DBus, label=unconfined),
+
+# Allow querying AppArmor
+dbus (send)
+    bus=session
+    path=/org/freedesktop/DBus
+    interface=org.freedesktop.DBus
+    member="GetConnectionAppArmorSecurityContext"
+    peer=(name=org.freedesktop.DBus, label=unconfined),
+
+# Allow binding the service to the requested connection name
+dbus (bind)
+    bus=session
+    name="core.ubuntu.media.Service",
+
+# Allow communications with unconfined processes
+dbus (receive, send)
+    bus=session
+    path=/com/ubuntu/media/Service{,/**}
+    interface=org.freedesktop.DBus{,.*}
+    peer=(label=unconfined),
+
+# Allow unconfined processes to introspect us
+dbus (receive)
+    bus=session
+    interface=org.freedesktop.DBus.Introspectable
+    peer=(label=unconfined),
+
+dbus (receive, send)
+    bus=session
+    path=/core/ubuntu/media/Service{,/**}
+    peer=(label=unconfined),
+
+# Allow sending mpris signals for session path
+# TODO: We send and receive TrackListReset, research why
+dbus (receive, send)
+    bus=session
+    path=/core/ubuntu/media/Service/sessions/**
+    interface="org.mpris.MediaPlayer2{,.Player,.TrackList}",
+
+# Allow sending properties signals for session path
+dbus (send)
+    bus=session
+    path=/core/ubuntu/media/Service/sessions/**
+    interface="org.freedesktop.DBus.Properties",
+
+# Allow access to powerd
+dbus (receive, send)
+    bus=system
+    path=/com/canonical/powerd
+    interface=com.canonical.powerd,
```

Comment:
This interface is used to access powerd (repowerd now) in Touch. If at some time that sw is snapped, it would make sense to create its own interface. I have removed this as apparently that has no adverse effects for media-hub.